packages/solutions/module-tools/src/builder/feature/redirect.ts

```diff
@@ -89,9 +92,31 @@ async function redirectImport(
         }
       }
 
+      if (redirect.autoExtension) {
+        if (ext === '') {
+          // add extension for relative path, no check if it's a directory.
+          if (jsExtension !== '.js' && name.startsWith('.')) {
+            str.overwrite(start, end, `${name}${jsExtension}`);
+            return;
+          }
+        }
+      }
+
+      // //  .c(m)ts -> .c(m)js
+      // // we don't want user use c(m)ts in source code directly
+      // if (isTsExt(name)) {
+      //   str.overwrite(
+      //     start,
+      //     end,
+      //     name.replace(/(.+)(\.[c|m]?)(ts)/, (_, p1, p2) => {
+      //       return `${p1}${p2}js`;
+      //     }),
+      //   );
```

Comment:
should we add this replace?  In fact,  I hope user don't use some import/export like `export * from './index.mts'`

Reply:
@chenjiahan

Reply:
Why users can't use `.mts`? It is possible to be used

Reply:
We must change [esbuild out extension](https://esbuild.github.io/api/#out-extension), it can not filter files, this request  your source file extension need be the same.

Reply:
Get, we can temporarily not support mts in this version.